test/activerecord-retry-reads_test.rb

```diff
@@ -23,50 +23,33 @@ module PG
 
 class TestActiveRecordRetryReads < Minitest::Test
   def test_abstract_module_has_methods_we_expect
-    abstract_module = ActiveRecord::ConnectionAdapters::DatabaseStatements
+    abstract_module = ActiveRecord::ConnectionAdapters::AbstractAdapter
     assert abstract_module.private_instance_methods.include?(:raw_execute)
-  end
-
-  def test_mysql2_module_has_methods_we_expect
-    mysql2_module = ActiveRecord::ConnectionAdapters::Mysql2::DatabaseStatements
-    assert mysql2_module.private_instance_methods.include?(:raw_execute)
 
-    parameters = mysql2_module.instance_method(:raw_execute).parameters
+    parameters = abstract_module.instance_method(:raw_execute).parameters
     assert_parameters_are_what_we_expect parameters
   end
 
-  def test_postgresql_module_has_methods_we_expect
-    postgresql_module = ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements
-    assert postgresql_module.instance_methods.include?(:raw_execute)
+  def test_adapter_modules_does_not_have_methods_we_expect
```

Comment:
What does this test protect us from?

Reply:
One of the adapters implementing their own `raw_execute` that doesn't call our monkey patch `super`. That seems unlikely, but I think it would be good to know if they add a `raw_execute` to any of them.